mlflow/tracking/client.py

```diff
@@ -1453,34 +1469,102 @@ def log_image(
+                will be stored as an artifact relative to the run's root directory (for
+                example, "dir/image.png"). This parameter is kept for backward compatibility
+                and should not be used together with `key`, `step`, or `timestamp`.
+            key: Image name for time-stepped image logging.
+            step: Integer training step (iteration) at which the image was saved.
+                Defaults to 0.
+            timestamp: Time when this image was saved. Defaults to the current system time.
 
         .. code-block:: python
-            :caption: Numpy Example
+            :caption: Time-stepped image logging numpy example
 
             import mlflow
             import numpy as np
 
             image = np.random.randint(0, 256, size=(100, 100, 3), dtype=np.uint8)
+            client = mlflow.MlflowClient()
+            run = client.create_run(experiment_id="0")
+            client.log_image(run.info.run_id, image, key="dogs", step=3)
 
+        .. code-block:: python
+            :caption: Time-stepped image logging pillow example
+
+            import mlflow
+            from PIL import Image
+
+            image = Image.new("RGB", (100, 100))
+            client = mlflow.MlflowClient()
+            run = client.create_run(experiment_id="0")
+            client.log_image(run.info.run_id, image, key="dogs", step=3)
+
+        .. code-block:: python
+            :caption: Legacy artifact file image logging numpy example
+
+            import mlflow
+            import numpy as np
+
+            image = np.random.randint(0, 256, size=(100, 100, 3), dtype=np.uint8)
+            client = mlflow.MlflowClient()
             run = client.create_run(experiment_id="0")
             client.log_image(run.info.run_id, image, "image.png")
 
         .. code-block:: python
-            :caption: Pillow Example
+            :caption: Legacy artifact file image logging pillow example
 
             import mlflow
             from PIL import Image
 
             image = Image.new("RGB", (100, 100))
-
+            client = mlflow.MlflowClient()
             run = client.create_run(experiment_id="0")
             client.log_image(run.info.run_id, image, "image.png")
         """
+        if artifact_file is not None and any(arg is not None for arg in [key, step, timestamp]):
+            raise TypeError(
+                "The `artifact_file` parameter cannot be used in conjunction with `key`, "
+                "`step`, or `timestamp` parameters. Please ensure that `artifact_file` is "
+                "specified alone, without any of these conflicting parameters."
+            )
+        elif artifact_file is None and key is None:
+            raise TypeError(
+                "Invalid arguments: Please specify exactly one of `artifact_file` or `key`. Use "
+                "`key` to log dynamic image charts or `artifact_file` for saving static images. "
+            )
+        elif artifact_file is not None:
```

Comment:
we can replace this elif check with if, since we raise exceptions for above 2 cases. It will be easier to read (arg check followed by actual handling)